### ISSUE TITLE
Question: Does the CDF have to be normalized?

### ISSUE THREAD

**Issue by mateisl** [tunnell/optimum_interval#1]
Hello,

I am currently trying your implementation for Yellins optimal interval method and was wondering if the CDF needs to be normalized in order for the algorithm to work.

Thanks in advance!

**Comment by tomeichlersmith** [comment]
I believe the CDF does have to be normalized such that the resulting energy cumulants are within the range [0,1] where the MC lookup table is defined. This is also implied by requiring the energy distribution $dN/dE$ to be normalized:

> The energy cumulant $\epsilon(E)$ is the fraction of energies expected below the energy $E$. Whatever the (1-normalized) energy distribution $dN/dE$, $dN/d\epsilon$ is uniform[0,1], where 0 and 1 correspond to the boundaries of our experimental range.

Oh lordy :facepalm: I just realized this is three years late

**Comment by tunnell** [comment]
Yes — the CDF has to be normalized so that it maps your full analysis range onto [0, 1]. @tomeichlersmith's answer above is exactly right; here's a bit more on the *why*, plus one point that trips people up.

**What "normalized" means here.** `spectrum_cdf` should be the cumulative of the signal *shape* $dN/dE$, normalized to 1 over the analysis window:

$$\epsilon(E) = \frac{\int_{E_\min}^{E} (dN/dE')\,dE'}{\int_{E_\min}^{E_\max} (dN/dE')\,dE'} \in [0,1],$$

so that $\epsilon(E_\min)=0$ and $\epsilon(E_\max)=1$.

**Why [0, 1] is required.** The method rests on the probability-integral transform: if the signal follows $dN/dE$, then $\epsilon(E)$ is Uniform[0, 1]. The Monte Carlo calibration is generated as uniform points on [0, 1] with the range endpoints 0 and 1 included, and interval "sizes" are measured as fractions of that unit interval. So your transformed events must live on that same [0, 1] with the range boundaries at exactly 0 and 1 — otherwise the observed interval sizes aren't comparable to the MC reference and the endpoints added for the real data land in the wrong place. (Concretely: if your CDF doesn't reach 1 at $E_\max$, the appended endpoint at 1 opens a phantom empty gap beyond your data and weakens/biases the limit; if it returns values outside [0, 1], the point list stops being sorted once 0 and 1 are appended and the gap sizes come out wrong.)

**The one gotcha:** normalize the *shape* to a CDF — do **not** fold the expected number of events $\mu$ into it. $\mu$ (the total expected signal count, $\propto$ cross section) is the parameter the method sets a limit on; it enters separately through the Poisson draw in the calibration, not through the CDF. Two different cross sections share the same `spectrum_cdf`; only a change of signal *shape* (e.g. a different WIMP mass) changes it.

I recently cleaned up and documented the repo — see [`EXPLANATION.md`](https://github.com/tunnell/optimum_interval/blob/master/EXPLANATION.md), section 2 ("The cumulant transform"), and there's now a `cumulant_points(events, spectrum_cdf)` helper that does the transform and adds the 0/1 endpoints for you.

Thanks @tomeichlersmith for fielding this one. Closing it out — and for the record, I'm only *two* years late to reply, not three, so by my accounting I'm actually ahead of schedule. 🙂
